ooyala_player/transcript.py

```diff
@@ -92,28 +94,45 @@ def _get_translations(self, selected_lang):
         except Exception as e:
             self.error = str(e.message)
         else:
-            for translation in translations_list:
-                if translation.get('state') == 'complete':
-                    lang_name = translation.get('target_language_name')
-                    lang_code = translation.get('target_language_iso_639_1_code')
-
-                    try:
-                        lang_info = get_language_info(lang_code)
-                        localized_name = lang_info.get('name_local')
-                    except KeyError:
-                        localized_name = lang_name
-
-                    self.translations.append({
-                        'language': lang_name,
-                        'lang_code': lang_code,
-                        'localized_name': localized_name,
-                        'selected': True if selected_lang in [lang_code, lang_name] else False,
-                        'url': TRANSLATION_DOWNLOAD_URL.format(
-                            project_id=self.project_id, transcript_file_id=self.transcript_id,
-                            translation_id=translation.get('id')
-                        ),
-                        'dir': 'rtl' if lang_name in RTL_LANGUAGES else 'ltr'
-                    })
+            if not translations_list:
+                language_api_endpoint = LANGUAGE_API_ENDPOINT.format(
+                    api_key_3play=self.api_key
+                )
+                try:
+                    response = urlopen(language_api_endpoint)
+                    data = json.loads(response.read())
+                    for language in data:
+                        if language.get('language_id') == self.language_id:
```

Comment:
@qazi-fahid I think you just need to create a method in this class, which would return the language details against a language code.
```python
     def get_language_details(self, lang_code):
            ...calls api     
```
You can simply use this method to get name of the language instead of hardcoded `English` details here:
https://github.com/edx-solutions/xblock-ooyala/blob/421baa2c8bc4ae44a770eb0de8db4c02b4bdcac3/ooyala_player/transcript.py#L40-L48

Reply:
done